WpfDesign.Designer/Project/Services/CreateComponentTool.cs

```diff
@@ -26,26 +26,31 @@
 
 namespace ICSharpCode.WpfDesign.Designer.Services
 {
-	/// <summary>
-	/// A tool that creates a component.
-	/// </summary>
-	public class CreateComponentTool : ITool
+    /// <summary>
+    /// A tool that creates a component.
+    /// </summary>
+    public class CreateComponentTool : ITool
 	{
 		protected ChangeGroup ChangeGroup;
 
 		readonly Type componentType;
-		MoveLogic moveLogic;
+        object[] arguments;
+
+        MoveLogic moveLogic;
 		Point createPoint;
-		
-		/// <summary>
-		/// Creates a new CreateComponentTool instance.
-		/// </summary>
-		public CreateComponentTool(Type componentType)
+
+		public event EventHandler<DesignItem> CreateComponentCompleted;
+
+        /// <summary>
+        /// Creates a new CreateComponentTool instance.
+        /// </summary>
+        public CreateComponentTool(Type componentType, object[] arguments)

```

Comment:
with this it is a breaking change.
Please create a additional constructor without the perameter

Reply:
sorry about that,done ser.

Reply:
the additional constructor can call the base, so we have no code duplicated

Reply:
Thank you for your thorough review.Had submited a new one